Sources/CorePayments/Networking/HTTPResponseParser.swift

```diff
@@ -4,32 +4,68 @@ import Foundation
 public class HTTPResponseParser {
     
     private let decoder: JSONDecoder
+    
+    // MARK: - Initializer
 
     public init(decoder: JSONDecoder = JSONDecoder()) { // exposed for test injection
         self.decoder = decoder
         decoder.keyDecodingStrategy = .convertFromSnakeCase
     }
     
-    // TODO: - Update this func (or file) to handle both GraphQL and REST error parsing
-    public func parse<T: Decodable>(_ httpResponse: HTTPResponse, as type: T.Type) throws -> T {
+    // MARK: - Public Methods
+
+    public func parseREST<T: Decodable>(_ httpResponse: HTTPResponse, as type: T.Type) throws -> T {
         guard let data = httpResponse.body else {
             throw APIClientError.noResponseDataError
         }
         
         if httpResponse.isSuccessful {
-            do {
-                let decodedData = try decoder.decode(T.self, from: data)
-                return (decodedData)
-            } catch {
-                throw APIClientError.jsonDecodingError(error.localizedDescription)
-            }
+            return try parseSuccessResult(data, as: T.self)
+        } else {
+            return try parseErrorResult(data, as: T.self)
+        }
```

Comment:
Is there ever a case where we couldn't parse the data into either an error or success result?

Reply:
The only case I can think of we wouldn't call this function at all, like for analytics. We aren't calling HTTPResponseParser at all from the TrackingEventsAPI, since we only care if it was a successful status or not.

(Example [here](https://github.com/paypal/paypal-ios/blob/networking-refactor/Sources/CorePayments/Networking/AnalyticsService.swift#L54) & [here](https://github.com/paypal/paypal-ios/blob/9aa4eb8031ffd646cbc4aef4343388d99a713494/Sources/CorePayments/Networking/TrackingEventsAPI.swift#L26))

Reply:
I went back and forth on the best way to write this file to be as DRY as possible and landed on this, but totally open to restructuring/ideas to make it more readable!

Reply:
Gotcha, I think how you have it makes sense! I was moreso curious about cases where we hit the else block but cannot parse the data into an error result. Not sure if that would ever happen though.